code/datums/gamemodes/crash.dm

```diff
@@ -103,8 +103,7 @@
 			X.amount_grown = X.max_grown
 		else // Handles Shrike etc
 			var/mob/living/carbon/xenomorph/X = i
-			X.upgrade_stored = X.xeno_caste.upgrade_threshold
-
+			X.hive?.upgrade_stored = X.xeno_caste.upgrade_threshold
```

Comment:
This might require some thought because a tier 2 should equal tier 1 + tier 2 requirements, etc for tier 3 since the hive evolution is shared like this now